best_practices/source_control/terra_source_control_cheatsheet.md

```diff
@@ -0,0 +1,256 @@
+# Source control on Terra: cheat sheet
+
+- [Source control on Terra: cheat sheet](#source-control-on-terra-cheat-sheet)
+  - [Check only code (no data) into source control](#check-only-code-no-data-into-source-control)
+    - [Exclude notebook outputs](#exclude-notebook-outputs)
+    - [Exclude data files](#exclude-data-files)
+  - [How to propagate notebook changes to source control](#how-to-propagate-notebook-changes-to-source-control)
+    - [(1) Use the Jupyter console to upload your GitHub SSH key and create an interactive terminal session](#1-use-the-jupyter-console-to-upload-your-github-ssh-key-and-create-an-interactive-terminal-session)
+    - [(2) Use the terminal to set up and clone your Git repository](#2-use-the-terminal-to-set-up-and-clone-your-git-repository)
+    - [(3) Check in your code changes](#3-check-in-your-code-changes)
+  - [FAQ](#faq)
+    - [How do I make changes to my branch?](#how-do-i-make-changes-to-my-branch)
+    - [I accidentally checked a new change to my main branch instead of a feature branch. How to fix this?](#i-accidentally-checked-a-new-change-to-my-main-branch-instead-of-a-feature-branch-how-to-fix-this)
+      - [How do I revert a modified file to its committed version?](#how-do-i-revert-a-modified-file-to-its-committed-version)
+
+This document summarizes some of the key practices described by the three-part document in this directory ([Part I](./terra_source_control_I.md), [Part II](./terra_source_control_II.md), [Part III](./terra_source_control_III.md)). It covers both [Terra.bio](https://app.terra.bio/) (Terra) workspaces and **All of Us Workbench** (AoU) workspaces.  In some cases there are differences between the two platforms.
+
+<!--It's primarily intended for analysts using source control for the materials within the featured workspaces, but these source control practices could be used by any AoU researcher who would like to use source control to manage their analysis code.-->
+
+## Check only code (no data) into source control
+
+### Exclude notebook outputs
+
+For the **All of Us Workbench**, the most important thing to remember is **notebooks checked into source control may not contain outputs**. Use [nbstripout](https://github.com/kynan/nbstripout) to remove all notebook outputs.
+This screencast demonstrates the use and working principles behind the `nbstripout` utility and how to use it as a Git filter:
+
+[![Img alt text](https://i.imgur.com/7oQHuJ5.png)](https://www.youtube.com/watch?v=BEMP4xacrVc)
+
+On the **All of Us Workbench**, if you are using Git from the terminal of your workspace,
+`nbstripout` is preinstalled and enabled. You can run `nbstripout --status` to confirm this.
+
+For Terra example workspaces, you may decide to retain some notebook outputs for explanatory value,
```

Comment:
Alternatively, the timing might be right to get nbstripout and the global gitignore into the Terra base Docker image. I'll check on that.

Reply:
TL;DR: lets go ahead and add the instructions for nbstripout and the global gitignore here for CWB

The way I [originally added support for that to AoU](https://github.com/DataBiosphere/terra-docker/pull/234/files#diff-df40bbb7ec38ec9c9fcdba2e7f8375b32309e761e675b54d5182f63b023b23b0) is no longer the right approach now that the PD is mounted at $HOME. It needs to be done via a startup script and it will take a bit of time to get that in place if folks want to proceed with that update.

Reply:
> Recommend removing this sentence, our guidance should err on the side of caution. (And less is more on this cheatsheet.)

Let's discuss this a bit further when we meet tomorrow— Given that so many cwb terra notebooks contain output, and it can be fine to (selectively) leave them in depending on context, I feel like it's going to confuse Terra users if we're too absolute.

Reply:
It turns out the `nbstripout` info already _was_ in part II (haha... I should have known that) and so I've linked to it.  (I added one bit to those part II instructions, which was to set `export PATH=$PATH:/home/jupyter/.local/bin` if necessary.  My tests show that this path is set for the default gatk VM but not in the Hail cluster terminal.)